contrib/scripts/unloaddb.sh

```diff
@@ -0,0 +1,402 @@
+#!/bin/bash
+#
+#
+#  Copyright 2016 CUBRID Corporation
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+verbose="no"   # set 'yes' for verbose mode
+num_proc=8
+table_size=()
+table_selected=()
+num_tables=0
+slot_selected=0
+slot_size=(0 0 0 0 0 0 0 0)
+num_tables_slot=(0 0 0 0 0 0 0 0)
+database=
+user="-u dba"
+pass=""
+total_pages=0
+from_file=""
+target_dir=$(pwd)
+
+slot=()
+
+function show_usage ()
+{
+         echo "Usage: $0 [OPTIONS] [TARGET]"
+         echo " OPTIONS"
+         echo "  -t arg  Set number of parallel process; default 8"
+         echo "  -i arg  input FILE of table names; default: dump all classes"
+         echo "  -u arg  Set database user name; default dba"
+         echo "  -p arg  Set dbuser password"
+         echo "  -D arg  Set directory for unloaddb output dir/files"
+         echo "  -v      Set verbose mode on"
+
+         echo ""
+         echo " EXAMPLES"
+         echo "  $0 -t 4 -v demodb          # unload all tables in demodb"
+         echo "  $0 -i file demodb          # unload tables listed in file in demodb"
+         echo "  $0 -u user1 -D /tmp -i file -t 4 -v demodb"
+         echo ""
+}
+
+#
+# Kill unloaddb processes in progress and delete the directory, and delete incomplete files
+# Directories/files created by the normally terminated unloaddb process is not deleted.
+#
+function cleanup ()
+{
+        local i
+        local pid
+        local exit_stat
+
+	echo "interrupted"
+
+        for ((i = 0; i < $num_proc; i++))
+        do
+                if [ -f $database.$i/unloaddb_$i.pid ];then
+                        pid=$(cat $database.$i/unloaddb_$i.pid)
+
+                        kill -0 $pid 2> /dev/null
+                        if [ $? -eq 0 ];then
+                                kill -9 $pid
+                                rm -rf $database.$i
+                        fi
+                fi
+        done
+}
+
+function get_password ()
+{
+	local password
+
+	read -sp "Enter Password : " password
+	echo $password
+	echo "" > /dev/tty
+}
+
+function verify_user_pass ()
+{
+	local msg
+	local USERNAME
+	local dbuser
+	local passwd
+	local qry_dba_grp="SELECT u.name FROM db_user AS u, TABLE(u.groups) AS g(x) where x.name = 'DBA'"
+	local retcode
+
+	dbuser=$(echo ${user} | cut -d' ' -f2-2)
+	USERNAME=$(echo ${dbuser} | tr [:lower:] [:upper:])
+
+	# check the database server is running
+	retcode=$(ps -ef | grep cub_server | grep $database | wc -l)
```

Comment:
If database has hostname with @ string such as demodb@localhost on HA mode, it can't be found the cub_server's pid from process list.
I think that you need to consider a hostname at database.

Reply:
Thank you for your comment, I didn't consider that.
Changed for check running db server excluding @hostname if it user gives it.
(demodb, demodb@localhost, demodb@ipaddr => only check for 'demodb' server is running)